lib/opal/nodes/literal.rb

```diff
@@ -241,6 +245,15 @@ def compile
 
         wrap '(', ')' if recv?
       end
+
+      def detect_and_remove_trailing_semicolon value, node
+        if value.match(/;\s*$/)
+          compiler.warning 'Xstr ends with semicolon', node.line
+          value.gsub(/;\s*$/, '')
```

Comment:
should be `sub` with `REGEXP_END` instead of `$` so it becomes `\z`

see https://github.com/opal/opal/blob/66849b5da5c54236eea5a0e5eaf95ba06a744fe3/lib/opal/regexp_anchors.rb

Reply:
Also this should just issue the warning as suggested by @iliabylich it's better not to touch xstr and let them be a tool that's both powerful and dangerous 😄